src/internal/sdata/convert.go

```diff
@@ -437,3 +438,14 @@ func parseTime(x string) (t time.Time, err error) {
 func isNullString(x string) bool {
 	return x == "null" || x == "nil" || len(x) == 0
 }
+
+// formatTimestampNTZ trims the "Z" at the end of a timestamp if the "Z" exists
+// this is assuming that the timestamp has no time zone
+// example: 2022-05-06T20:18:10Z -> 2022-05-06T20:18:10
```

Comment:
I'd add a note explaining why we have to mutate a string instead of just formatting it correctly the first time.  (Because we don't know the type, and database/sql converts it to a time.Time before we can even get it as a string.)